### ISSUE TITLE
BYO token BaseDAO extension

### ISSUE THREAD

**Issue by michaeljklein** [tqtezos/baseDAO#170]
# Clarification and motivation

Requirement: allow holders of a pre-existing token contract to lock up (i.e. wrap) their tokens as DAO tokens inside of the BaseDAO contract.

Example: 

As a token issuer of an FA2 token called `DROP`, I want to create a BaseDAO in which `DROP` can be wrapped as `bDROP` to enable participation in the DAO.

# Acceptance criteria

## Changes to BaseDAO

The (un)freeze entrypoint will perform a FA2 transfer to/from a fixed existing, external, FA2 governance token.

In other words, wrapping `N` tokens will:
1) mint N wrapped (frozen) tokens on the DAO side and
2) transfer N governance tokens from the gov-FA2 to the DAO.

Unwrapping N tokens likewise transfers N governance tokens to the gov-FA2 from the DAO and burns N wrapped tokens.

These entrypoints will replace the explicit freeze/unfreeze entrypoints specified here: https://github.com/tqtezos/baseDAO/issues/140

The DAO contract will no longer internally account for an unfrozen governance token.
